sql/catalyst/src/main/scala/org/apache/spark/sql/types/UserDefinedType.scala

```diff
@@ -74,7 +74,7 @@ abstract class UserDefinedType[UserType] extends DataType with Serializable {
   /**
    * The default size of a value of the UserDefinedType is 4096 bytes.
    */
-  override def defaultSize: Int = 4096
+  override def defaultSize: Int = sqlType.defaultSize
```

Comment:
The old value is bad guess, because we always materialize the rows as sqlType, so should use that. Will remove the comment while merging.